gocd/gocd.go

```diff
@@ -0,0 +1,210 @@
+		Approval              struct {
+			Type          string `json:"type"`
+			Authorization struct {
+				Roles []interface{} `json:"roles"`
+				Users []interface{} `json:"users"`
+			} `json:"authorization"`
+		} `json:"approval"`
+		EnvironmentVariables []interface{} `json:"environment_variables"`
+		Jobs                 []struct {
+			Name                 string        `json:"name"`
+			RunInstanceCount     interface{}   `json:"run_instance_count"`
+			Timeout              interface{}   `json:"timeout"`
+			EnvironmentVariables []interface{} `json:"environment_variables"`
+			Resources            []interface{} `json:"resources"`
+			Tasks                []struct {
+				Type       string `json:"type"`
+				Attributes struct {
+					RunIf            []string    `json:"run_if"`
+					OnCancel         interface{} `json:"on_cancel"`
+					Command          string      `json:"command"`
+					Arguments        []string    `json:"arguments"`
+					WorkingDirectory string      `json:"working_directory"`
+				} `json:"attributes"`
+			} `json:"tasks"`
+			Tabs       []interface{} `json:"tabs"`
+			Artifacts  []interface{} `json:"artifacts"`
+			Properties interface{}   `json:"properties"`
+		} `json:"jobs"`
+	} `json:"stages"`
+	TrackingTool interface{} `json:"tracking_tool"`
+	Timer        interface{} `json:"timer"`
+}
+
+func unmarshalPipeline(data []byte) (pipeline Pipeline, err error) {
+	err = json.Unmarshal(data, &pipeline)
+	if err != nil {
+		return pipeline, err
+	}
+	return pipeline, nil
+}
+
+func unmarshalPipelineConfig(data []byte) (pipelineConfig PipelineConfig, err error) {
+	err = json.Unmarshal(data, &pipelineConfig)
+	if err != nil {
+		return pipelineConfig, err
+	}
+	return pipelineConfig, nil
+}
+
+func ReadPipelineJSONFromFile(path string) (pipeline Pipeline, err error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return pipeline, err
+	}
+	return unmarshalPipeline(data)
+}
+
+func pipelineConfigPOST(gocdULR string, pipelineConfig PipelineConfig) (pipeline Pipeline, err error) {
+
+	// Generated by curl-to-Go: https://mholt.github.io/curl-to-go
+
+	payloadBytes, err := json.Marshal(pipelineConfig)
+	if err != nil {
+		return pipeline, err
+	}
+	payloadBody := bytes.NewReader(payloadBytes)
+
+	req, err := http.NewRequest("POST", gocdULR+"/go/api/admin/pipelines", payloadBody)
+	if err != nil {
+		return pipeline, err
+	}
+	user := os.Getenv("GOCDUSER")
+	pass := os.Getenv("GOCDPASSWORD")
+	req.SetBasicAuth(user, pass)
+	req.Header.Set("Accept", "application/vnd.go.cd.v1+json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return pipeline, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return pipeline, err
+	}
+
+	var prettyJSON bytes.Buffer
+	err = json.Indent(&prettyJSON, body, "", "\t")
+	if err != nil {
+		return pipeline, err
+	}
+	//fmt.Println("pipelineConfigPOST:", string(prettyJSON.Bytes()))
```

Comment:
Preferable to using comments, we should be able to run in a formal "debug" or "verbose" mode.  Look into Logrus or  glog?